design/general-global-rate-limit-policy.md

```diff
@@ -0,0 +1,100 @@
+# General Global RateLimit Policy
+
+Status: Draft
+
+## Abstract
+Define a general rate limit policy in the Contour configuration to be used as a global rate limit policy by all HTTPProxy.
+
+## Background
+See https://github.com/projectcontour/contour/issues/5357 for context.
+
+## Goals
+- Define a general L7 global rate limit policy which is used by all HTTPProxy on the `VirtualHost` level.
+
+## Non Goals
+- General rate limit policy for Route level `rateLimitPolicy`.
+- L4 rate limiting support. 
+
+## High-Level Design
+Currently, the global `rateLimitPolicy` for `VirtualHost` is set per HTTPProxy. This option allows each service to configure its own global `rateLimitPolicy`. However, for a single tenant setup/general usecase, this means you have to edit all the HTTPProxy objects with the same ratelimit policies.
+
+A new field, `GeneralRateLimitPolicy` (optional), will be added to `RateLimitServiceConfig` which is part of `ContourConfigurationSpec`. This field will define rate limit descriptors that will be added to every `VirtualHost`.
```

Comment:
Personally I'd prefer "default rate limit policy" as the name for this. Also need to decide if we want to include "global" as part of the name, to differentiate from local rate limiting which is not affected by this proposal. `DefaultGlobalRateLimitPolicy` ends up being a  pretty long name though; open to ideas here.

Reply:
Ah!! thank you for choosing a good name for this 😄 
I will use that actually. I was struggling to come up with a better word.